### ISSUE TITLE
When reading gdx file, "TypeError: object of type 'NoneType' has no len()"

### ISSUE THREAD

**Issue by mmowers** [khaeru/py-gdx#29]
Thanks for this package! I'm looking forward to using it! First I have to figure out this issue though.

After installing py-gdx, on the first attempt to read a gdx file, I get a typeError:
```
(py34) C:\Users\Owner\Python Libs\py-gdx speed test>python
Python 3.4.5 |Continuum Analytics, Inc.| (default, Jul  5 2016, 14:53:07) [MSC v.1600 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from gdx import File
>>> f = File("LDC_static_inputs.gdx")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\site-packages\gdx-3-py3.4.egg\gdx\__init__.py", line 62, in __init__
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\site-packages\gdx-3-py3.4.egg\gdx\api.py", line 80, in __init__
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\site-packages\gdx-3-py3.4.egg\gdx\api.py", line 53, in _gams_dir
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\ntpath.py", line 253, in dirname
    return split(p)[0]
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\ntpath.py", line 217, in split
    d, p = splitdrive(p)
  File "C:\Users\Owner\Anaconda3\envs\py34\lib\ntpath.py", line 159, in splitdrive
    if len(p) > 1:
TypeError: object of type 'NoneType' has no len()
>>>
```
The file "LDC_static_inputs.gdx" is in C:\Users\Owner\Python Libs\py-gdx speed test.

I am using Windows 10 home Version 1607 Build 14393.693.

I wonder if this is related to an installation issue I had. When running `python setup.py install` in py-gdx/, I had an error that said `Setup script exited with error: Microsoft Visual Studio C++ 10.0 is required` during the installation of pandas. I solved by just installing numpy and pandas directly with `conda install numpy` and `conda install pandas`, and then running `python setup.py install` again.

Any help would be much appreciated.

Thanks!

**Comment by khaeru** [comment]
Could you please execute the following and reply with the output?

```python
from shutil import which
which('gams')
```

This is supposed to give the path to GAMS on your system, e.g. `C:\blah\blah\GAMS24.6\gams.exe`.

As the last three lines of the traceback show, something is happening in the file `nypath.py`, specifically in `splitdrive()`, which is [this code in Python 3.4](https://github.com/python/cpython/blob/3.4/Lib/ntpath.py#L136) and [this code in Python 3.6](https://github.com/python/cpython/blob/3.6/Lib/ntpath.py#L118).

This is part of Python, not PyGDX; and specifically the Windows version of the Python `path` standard library. Our automated tests only run on Linux, which is probably why I've never seen this before!

The output of `which('gams')` seems to be giving `splitdrive()` some trouble. Since `splitdrive()` seems to have changed from Python 3.4 to Python 3.6, maybe updating your Python distribution would fix the problem. But I'd still like to give an intelligible error message in this case or mention it in the documentation.

**Comment by mmowers** [comment]
Looks like I'm getting "None" when i print which('gams').

```
Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

C:\Users\Owner>activate py34

(py34) C:\Users\Owner>python
Python 3.4.5 |Continuum Analytics, Inc.| (default, Jul  5 2016, 14:53:07) [MSC v.1600 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from shutil import which
>>> which('gams')
>>> print(which('gams'))
None
>>> exit()

(py34) C:\Users\Owner>deactivate py34

C:\Users\Owner>python
Python 3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from shutil import which
>>> print(which('gams'))
None
>>> exit()

C:\Users\Owner>
```

I checked my packages for the py34 environment, and GAMS is definitely in there:
```
C:\Users\Owner>activate py34

(py34) C:\Users\Owner>conda list
# packages in environment at C:\Users\Owner\Anaconda3\envs\py34:
#
backports.shutil-which    3.5.1                     <pip>
cfgmcc                    1                         <pip>
dctmcc                    1                         <pip>
future                    0.16.0                    <pip>
GAMS                      1.0                       <pip>
gamsxcc                   1                         <pip>
gdxcc                     7                         <pip>
gevmcc                    6                         <pip>
gmdcc                     2                         <pip>
gmomcc                    15                        <pip>
mkl                       2017.0.1                      0
numpy                     1.11.3                   py34_0
optcc                     2                         <pip>
pandas                    0.19.2              np111py34_1
pip                       9.0.1                    py34_1
python                    3.4.5                         0
python-dateutil           2.6.0                    py34_0
pytz                      2016.10                  py34_0
setuptools                27.2.0                   py34_1
six                       1.10.0                   py34_0
vs2010_runtime            10.00.40219.1                 2
wheel                     0.29.0                   py34_0
xarray                    0.9.1                     <pip>

(py34) C:\Users\Owner>python
Python 3.4.5 |Continuum Analytics, Inc.| (default, Jul  5 2016, 14:53:07) [MSC v.1600 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from shutil import which
>>> print(which('GAMS'))
None
```

**Comment by khaeru** [comment]
Where is GAMS installed on your system? Can you type simply `gams` at the Windows command prompt and paste the results?

**Comment by mmowers** [comment]
Aha! I didn't have gams added to PATH. Everything works now. FYI, if you want this for your documentation, here are the things that tripped me up:

1. When running `python setup.py install` in py-gdx/, I had an error that said `Setup script exited with error: Microsoft Visual Studio C++ 10.0 is required` during the installation of pandas. I solved by just installing numpy and pandas directly with `conda install numpy` and `conda install pandas`, and then running `python setup.py install` again.
2. I had to make sure gams was added to PATH. I reinstalled and followed instructions here to make sure gams was added to PATH: http://www.gams.com/latest/docs/userguides/userguide/_u_g__w_i_n__i_n_s_t_a_l_l.html#CLI

Thanks again!
